packages/aila/src/protocol/schemas/quiz/conversion/rawQuizIngest.ts

```diff
@@ -1,135 +1,233 @@
-    format: z.enum(["png", "jpg", "jpeg", "webp", "gif", "svg"]).optional(),
-    secureUrl: z.string().url(),
-    url: z.string().url().optional(),
-    height: z.number().optional(),
-    width: z.number().optional(),
-    metadata: z.union([
-      z.array(z.any()),
-      z.object({
-        attribution: z.string().optional(),
-        usageRestriction: z.string().optional(),
-      }),
-    ]),
-    publicId: z.string().optional(),
-    version: z.number().optional(),
-  }),
-  type: z.literal("image"),
-});
-
-export type StemImageObject = z.infer<typeof stemImageObjectSchema>;
-
-export type StemObject = StemTextObject | StemImageObject;
-
-const answersSchema = z.object({
-  "multiple-choice": z.array(multipleChoiceSchema).nullable().optional(),
-  match: z.array(matchSchema).nullable().optional(),
-  order: z.array(orderSchema).nullable().optional(),
-  "short-answer": z.array(shortAnswerSchema).nullable().optional(),
-  "explanatory-text": z.null().optional(),
-});
-
-export const rawQuizQuestionSchema = z.object({
-  questionId: z.number(),
-  questionUid: z.string(),
-  questionType: z.enum([
-    "multiple-choice",
-    "match",
-    "order",
-    "short-answer",
-    "explanatory-text",
-  ]),
-  questionStem: z
-    .array(z.union([stemTextObjectSchema, stemImageObjectSchema]))
-    .min(1),
-  answers: answersSchema.nullable().optional(),
-  feedback: z.string(),
-  hint: z.string(),
-  active: z.boolean(),
-});
-
-export const rawQuizSchema = z
-  .array(rawQuizQuestionSchema)
-  .nullable()
-  .optional();
-
-export type RawQuiz = ConvertKeysToCamelCase<z.infer<typeof rawQuizSchema>>;
-export type QuizProps = {
-  questions: NonNullable<RawQuiz>;
-  imageAttribution: { attribution: string; questionNumber: string }[];
-  isMathJaxLesson: boolean;
-};
-
-export function keysToCamelCase<T>(obj: T): ConvertKeysToCamelCase<T> {
-  if (Array.isArray(obj)) {
-    return obj.map((item) =>
-      keysToCamelCase(item as unknown[]),
-    ) as ConvertKeysToCamelCase<T>;
-  } else if (obj && typeof obj === "object") {
-    return Object.entries(obj).reduce((acc, [key, value]) => {
-      const newKey = convertKey(key);
-      acc[newKey as keyof typeof acc] = keysToCamelCase(value);
-      return acc;
-    }, {} as ConvertKeysToCamelCase<T>);
+
+/**
+ * Convert raw quiz from Oak curriculum format to Quiz V2 format
+ */
+export function convertRawQuizToV2(rawQuiz: RawQuiz): QuizV2 {
+  if (!rawQuiz || !Array.isArray(rawQuiz)) {
+    return {
+      version: "v2",
+      questions: [],
+      imageAttributions: [],
+    };
   }
-  return obj as ConvertKeysToCamelCase<T>;
+
+  // Collect all image attributions from all questions
+  const allImageAttributions: Array<{ imageUrl: string; attribution: string }> =
+    [];
+
+  const questions = rawQuiz
+    .filter((rawQuestion) => rawQuestion.question_type !== "explanatory-text")
+    .map((rawQuestion): QuizV2Question => {
+      if (!rawQuestion) {
+        // Fallback for invalid questions
```

Comment:
Same as below, question I guess around when we think invalid questions might arise and whether we'd be better to filter them out and report an error, or to fail hard.

Reply:
Yep, agree